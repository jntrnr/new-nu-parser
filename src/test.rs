use crate::resolver::Resolver;
use crate::{compiler::Compiler, parser::Parser};
use std::path::Path;

fn evaluate_example(fname: &Path) -> String {
    let mut compiler = Compiler::new();
    let contents = std::fs::read(fname).expect("We only run tests found by glob");

    // Homogenize line endings from CRLF to LF to make sure tests work both on Windows and Linux/macOS
    let contents = String::from_utf8(contents)
        .unwrap()
        .replace("\r\n", "\n")
        .as_bytes()
        .to_owned();

    let span_offset = compiler.span_offset();
    compiler.add_file(&fname.to_string_lossy(), &contents);

    let parser = Parser::new(compiler, span_offset);
    compiler = parser.parse();

    let mut result = compiler.display_state();

    if !compiler.errors.is_empty() {
        return result;
    }

    let mut resolver = Resolver::new(&compiler);
    resolver.resolve();
    result.push_str(&resolver.display_state());

    compiler.merge_name_bindings(resolver.to_name_bindings());

    result
}

#[test]
fn test_node_output() {
    insta::glob!("../tests", "*.nu", |path| {
        insta::assert_snapshot!(evaluate_example(path));
    });
}
