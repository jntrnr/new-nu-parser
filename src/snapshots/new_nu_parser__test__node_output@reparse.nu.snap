---
source: src/test.rs
expression: evaluate_example(path)
input_file: tests/reparse.nu
---
0: Name (0 to 3)
1: Name (4 to 7)
2: Name (9 to 10)
3: Garbage (10 to 11)
4: Name (12 to 18)
5: List([NodeId(2), NodeId(3), NodeId(4)]) (8 to 18)
6: Name (22 to 26)
7: Garbage (30 to 31)
8: Name (33 to 36)
9: Int (37 to 38)
10: Block(BlockId(0)) (22 to 38)
11: Garbage (38 to 38)
12: Closure { params: None, block: NodeId(10) } (20 to 38)
13: Block(BlockId(1)) (0 to 38)

